ofdb-gateways/src/mailgun.rs

```diff
@@ -0,0 +1,66 @@
+use ofdb_entities::{event::*, place::*};
+
+// TODO: move into ofdb-core crate
+/// The kind of notification.
+#[derive(Debug, PartialEq)]
+pub enum Notification {
+    PlaceAdded(Place),
+    PlaceUpdated(Place),
+    EventAdded(Event),
+    EventUpdated(Event),
+}
+
+// TODO: move into ofdb-core crate
+/// A subsystem that manages everything concerning
+/// notifications.
+pub trait NotificationManager {
+    fn notify(&self, n: Notification);
+}
+
+/// A email notification manager.
+pub struct Manager {
+    pub api_key: String,
+    pub domain: String,
+    pub from_email: String,
+    pub to_email: String,
+}
+
+impl NotificationManager for Manager {
+    fn notify(&self, n: Notification) {
+        match n {
+            Notification::PlaceAdded(ref _place) => {
+                let client = reqwest::Client::new();
+                let params = [
+                    ("from", &self.from_email),
+                    ("to", &self.to_email),
+                    // TODO: ("subject", SUBJECT TEXT FROM PLACE ),
+                    // TODO: ("text", TEXT FROM PLACE),
+                ];
+                let res = client
+                    .post(&self.api_url())
+                    .form(&params)
+                    .basic_auth("api", Some(&self.api_key))
+                    .send();
+                match res {
+                    Ok(res) => {
+                        if res.status().is_success() {
+                            debug!("Mail provider response: {:#?}", res);
+                        } else {
+                            error!("Could not send email: response status: {:?}", res.status());
```

Comment:
We should distinguish between a technical failure (Err, see below) and an error response (here). Both log messages start with the same text.